test/service/tests/integration_test.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use cumulus_primitives::ParaId;
+use cumulus_test_service::initial_head_data;
+use futures::join;
+use sc_service::TaskExecutor;
+use substrate_test_runtime_client::AccountKeyring::*;
+
+#[substrate_test_utils::test]
+#[ignore]
+async fn integration_test(task_executor: TaskExecutor) {
```

Comment:
Maybe a more expressive name what this is doing? :D

Reply:
Done in 56722dbcaf8c23c1bba7954292ca30b76a2c1a4e

"test_collating_and_non_collator_mode_catching_up"